modules/appeals_api/app/services/appeals_api/request_validator.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  class RequestValidator
+    include SentryLogging
+
+    class EvidenceSubmissionRequestValidationError < StandardError; end
+
+    def initialize(notice_of_disagreement, veteran_identifier)
+      @notice_of_disagreement = notice_of_disagreement
+      @veteran_id = veteran_identifier
+    end
+
+    def call
+      return [:error, veteran_id_error] unless veteran_ids_match?
+      return [:error, review_option_error] unless evidence_submission_selected?
+      return [:error, expired_submission_window?] if expired_submission_window?
+
+      [:ok, {}]
+    end
+
+    private
+
+    def veteran_ids_match?
+      @notice_of_disagreement.auth_headers['X-VA-Veteran-SSN'] == @veteran_id
+    end
+
+    def evidence_submission_selected?
+      @board_review_option ||= @notice_of_disagreement.form_data['data']['attributes']['boardReviewOption']
+      @board_review_option == 'evidence_submission'
+    end
+
+    def expired_submission_window?
+      # do the thing to check that status == success and within 90-day window
+      true
+    end
+
+    def veteran_id_error
+      {
+        title: 'Veteran Id Mismatch',
+        detail: log_error(:veteran_id_mismatch),
+        meta: { provided_veteran_id: @veteran_id, provided_nod_id: @notice_of_disagreement.id }
+      }
+    end
+
+    def review_option_error
+      {
+        title: 'Unsupported Board Review Option',
+        detail: log_error(:unsupported_review_option),
+        meta: { selected_board_review_option: @board_review_option }
+      }
+    end
+
+    def expired_submission_error
+      {
+        title: 'Expired submission window',
+        detail: log_error(:expired_submission_window),
+        meta: { nod_date_submitted: '01/17/2020' }
```

Comment:
I like adding the meta info! Feels helpful to the consumer to know the original submission date of the NOD. 👍 

I know it's a draft, but just want to point out that this shouldn't be a hard-coded date when this is finished :) 

Let's double check whether other dates are returned in any sort of standard format in other APIs and follow that (which may just end up being what you already have :) ).

Reply:
Haven't solved for this yet :) It's a placeholder. I don't even know that we're going to check against this in this iteration.